baremetal/metalmachinetemplate_manager_test.go

```diff
@@ -0,0 +1,212 @@
+								"cluster.x-k8s.io/cloned-from-groupkind": "Metal3MachineTemplate.infrastructure.cluster.x-k8s.io",
+							},
+						},
+						Spec: capm3.Metal3MachineSpec{
+							AutomatedCleaningMode: "metadata",
+						},
+					},
+					{
+						TypeMeta: metav1.TypeMeta{},
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "machine-3",
+							Namespace: "foo",
+							Annotations: map[string]string{
+								"cluster.x-k8s.io/cloned-from-name":      "abc",
+								"cluster.x-k8s.io/cloned-from-groupkind": "Metal3MachineTemplate.infrastructure.cluster.x-k8s.io",
+							},
+						},
+						Spec: capm3.Metal3MachineSpec{
+							AutomatedCleaningMode: "metadata",
+						},
+					},
+				},
+			},
+		}),
+		Entry("Disk cleaning enabled", testCaseUpdate{
+			ExpectedValue: "metadata",
+			M3MachineTemplate: &capm3.Metal3MachineTemplate{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: capm3.GroupVersion.String(),
+					Kind:       "Metal3MachineTemplate",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "abc",
+					Namespace: "foo",
+				},
+				Spec: capm3.Metal3MachineTemplateSpec{
+					Template: capm3.Metal3MachineTemplateResource{
+						Spec: capm3.Metal3MachineSpec{
+							AutomatedCleaningMode: "metadata",
+						},
+					},
+				},
+			},
+			M3MachineList: &capm3.Metal3MachineList{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: capm3.GroupVersion.String(),
+					Kind:       "Metal3MachineList",
+				},
+				ListMeta: metav1.ListMeta{},
+				Items: []capm3.Metal3Machine{
+					{
+						TypeMeta: metav1.TypeMeta{},
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "machine-1",
+							Namespace: "foo",
+							Annotations: map[string]string{
+								"cluster.x-k8s.io/cloned-from-name":      "abc",
+								"cluster.x-k8s.io/cloned-from-groupkind": "Metal3MachineTemplate.infrastructure.cluster.x-k8s.io",
+							},
+						},
+						Spec: capm3.Metal3MachineSpec{
+							AutomatedCleaningMode: "disabled",
+						},
+					},
+					{
+						TypeMeta: metav1.TypeMeta{},
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "machine-2",
+							Namespace: "foo",
+							Annotations: map[string]string{
+								"cluster.x-k8s.io/cloned-from-name":      "abc",
+								"cluster.x-k8s.io/cloned-from-groupkind": "Metal3MachineTemplate.infrastructure.cluster.x-k8s.io",
+							},
+						},
+						Spec: capm3.Metal3MachineSpec{
+							AutomatedCleaningMode: "disabled",
+						},
+					},
+					{
+						TypeMeta: metav1.TypeMeta{},
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "machine-3",
+							Namespace: "foo",
+							Annotations: map[string]string{
+								"cluster.x-k8s.io/cloned-from-name":      "abc",
+								"cluster.x-k8s.io/cloned-from-groupkind": "Metal3MachineTemplate.infrastructure.cluster.x-k8s.io",
+							},
+						},
+						Spec: capm3.Metal3MachineSpec{
+							AutomatedCleaningMode: "disabled",
+						},
+					},
+				},
+			},
+		}),
+	)
+})
```

Comment:
This seems to be copy of the previous [file](https://github.com/metal3-io/cluster-api-provider-metal3/blob/974a7fb98b0f5686d8cd9854b685c09f39de026d/baremetal/metal3machinetemplate_manager.go)?

Reply:
Ops, two copies of the test code. Good catch